### PR TITLE
Bnb/ws power law

### PR DESCRIPTION
**Pull request by bnb32**

### CODE REVIEW

**Comment by bnb32** [comment]
@grantbuster Main edits in `feature_handling.py`, `interpolation.py`, and `test_data_handling_nc.py`

File: sup3r/preprocessing/feature_handling.py [hunk 1]
```diff
@@ -724,6 +725,98 @@ def compute(cls, data, height):
         return rews
 
 
+class UWindPowerLaw(DerivedFeature):
+    """U wind component feature class with needed inputs method and compute
+    method. Uses power law extrapolation to get values above surface
+
+    https://en.wikipedia.org/wiki/Wind_profile_power_law
```
Comment (grantbuster) [inline sup3r/preprocessing/feature_handling.py:732]: Thoughts on using this NOAA reference and set alpha=0.2 and put alpha as a class variable for transparency?

https://csl.noaa.gov/projects/lamar/windshearformula.html

File: tests/data/test_era5_co_2012.nc
Comment (grantbuster) [inline tests/data/test_era5_co_2012.nc:1]: Just a reminder to be really intentional when adding big test files like this. Every file added is added forever and bloats the repo size.

File: tests/data_handling/test_data_handling_nc_cc.py [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+    DataHandlerNCforCCwithPowerLaw,
+)
+
+
+def test_data_handling_nc_cc_power_law():
+    """Make sure the power law extrapolation of wind operates correctly"""
+    input_files = [os.path.join(TEST_DATA_DIR, 'uas_test.nc')]
+
+    with xr.open_mfdataset(input_files) as fh:
+        u_100m = fh['uas'].values * 100**0.143
```
Comment (grantbuster) [inline tests/data_handling/test_data_handling_nc_cc.py:22]: if you set alpha to class attr you can reference that here

**Review by grantbuster — CHANGES_REQUESTED** [review]
minor thought on power law reference and alpha value

File: sup3r/preprocessing/feature_handling.py (hunk 1 above)
Comment (bnb32) [reply sup3r/preprocessing/feature_handling.py:732]: Class var is a good idea. Why the disagreement between the alpha values though?

File: tests/data/test_era5_co_2012.nc
Comment (bnb32) [reply tests/data/test_era5_co_2012.nc:1]: Yeah this is trimmed for only needed vars but I misnamed it last merge.

File: sup3r/preprocessing/feature_handling.py (hunk 1 above)
Comment (grantbuster) [reply sup3r/preprocessing/feature_handling.py]: I'm not sure but NOAA is more defensible haha

Comment (grantbuster) [reply sup3r/preprocessing/feature_handling.py]: our high school teachers would be so proud

**Review by grantbuster — APPROVED** [review]

File: sup3r/preprocessing/feature_handling.py (hunk 1 above)
Comment (bnb32) [reply sup3r/preprocessing/feature_handling.py]: Our high school teachers were wrong. Wikipedia actually links to a paper. NOAA doesn't have a source for 0.2 that I can see?

**Pull request merged**